hadoop-hdds/docs/content/concept/RocksDB.md

```diff
@@ -0,0 +1,161 @@
+    *   `keyContainerTable`: Maps keys to the containers they are in.
+    *   `containerKeyCountTable`: Stores the number of keys in each container.
+    *   `replica_history_v2`: Tracks the historical locations of container replicas with BCSID, which is essential for auditing and diagnostics.
+
+## 3. Tunings applicable to RocksDB
+
+Effective tuning of RocksDB can significantly impact Ozone's performance. Ozone exposes several configuration properties to tune RocksDB behavior. These properties are typically found in `ozone-default.xml` and can be overridden in `ozone-site.xml`.
+
+### General Settings
+
+Ozone provides a set of general RocksDB configurations that apply to all services (OM, SCM, and Datanodes) unless overridden by more specific settings. With the exception of `hdds.db.profile` and `ozone.metastore.rocksdb.cf.write.buffer.size`, these properties are defined in `RocksDBConfiguration.java`.
+
+*   `hdds.db.profile`: Specifies the RocksDB profile to use, which determines the default `DBOptions` and `ColumnFamilyOptions`. Default value: `DISK`.
+    *   Possible values include `SSD` and `DISK`.
+    *   For example, setting this to `SSD` will apply tunings optimized for SSD storage.
+
+*   **Write Options:**
+    *   `hadoop.hdds.db.rocksdb.writeoption.sync`: If set to `true`, writes are synchronized to persistent storage, ensuring durability at the cost of performance. If `false`, writes are flushed asynchronously. Default: `false`.
+
+*   `ozone.metastore.rocksdb.cf.write.buffer.size`: The write buffer (memtable) size for each column family of the rocksdb store. Default: `128MB`.
+
+*   **Write-Ahead Log (WAL) Management:**
+    *   `hadoop.hdds.db.rocksdb.WAL_ttl_seconds`: The time-to-live for WAL files in seconds. Default: `1200`.
+    *   `hadoop.hdds.db.rocksdb.WAL_size_limit_MB`: The total size limit for WAL files in megabytes. When this limit is exceeded, the oldest WAL files are deleted. A value of `0` means no limit. Default: `0`.
+
+*   **Logging:**
+    *   `hadoop.hdds.db.rocksdb.logging.enabled`: Enables or disables RocksDB's own logging. Default: `false`.
+    *   `hadoop.hdds.db.rocksdb.logging.level`: The logging level for RocksDB (INFO, DEBUG, WARN, ERROR, FATAL). Default: `INFO`.
+    *   `hadoop.hdds.db.rocksdb.max.log.file.size`: The maximum size of a single RocksDB log file. Default: `100MB`.
+    *   `hadoop.hdds.db.rocksdb.keep.log.file.num`: The maximum number of RocksDB log files to retain. Default: `10`.
+
+### Ozone Manager (OM) Specific Settings
+
+These settings, defined in `ozone-default.xml`, apply specifically to the Ozone Manager.
+
+*   `ozone.om.db.max.open.files`: The total number of files that a RocksDB can open in the OM. Default: `-1` (unlimited).
+*   `ozone.compaction.service.enabled`: Enable or disable a background job that periodically compacts rocksdb tables flagged for compaction. Default: `false`.
```

Comment:
it should really be named ozone.om.compaction.service.enabled. Opened https://issues.apache.org/jira/browse/HDDS-13525 to fix it.

Reply:
HDDS-13525 is merged and so updated this PR to reflect the change.